### DISCUSSION TITLE
Full-text search for a field

### DISCUSSION THREAD

**Discussion by ava57r** [qdrant/qdrant#3293] [General]
Hello.
As far as I know, the index for fields with full-text search is stored only in RAM.

how do you estimate memory consumption for full-text search?
Are there any recommendations for setting up such fields to reduce memory consumption?

**Comment by ava57r** [comment]
I try to upload data into a collection
This collection has some fields with full-text params
And I can not upload data into a collection more than fixed amount vectors
I tried to use 
SQ
PQ x8
PQ x16
PQ x32

I always get OOM.

**Reply by generall** [reply]
We are working on making full-text index more compact in the next minor release of Qdrant. 

> how do you estimate memory consumption for full-text search?

That depends on the volume of text you are indexing, but I don't have any benchmarks for now

**Comment by ava57r** [comment]
How does index RAM depend on `min_token_len` and `max_token_len`?

Thanks

**Reply by generall** [reply]
smaller `min_token_len`  = more ram usage
higher `max_token_len` = more ram usage
